src/main.c

```diff
@@ -42,7 +48,30 @@ int main(int argc, char **argv)
 	while (true) {
 		volatile exception_s e;
 		TRY_CATCH (e, EXCEPTION_ALL) {
-			gdb_main();
+			SET_IDLE_STATE(false);
+			while (gdb_target_running && cur_target) {
+				gdb_poll_target();
+
+				// Check again, as `gdb_poll_target()` may
+				// alter these variables.
+				if (!gdb_target_running || !cur_target)
+					break;
+				char c = gdb_if_getchar_to(0);
+				if (c == '\x03' || c == '\x04')
+					target_halt_request(cur_target);
+				platform_pace_poll();
+#ifdef ENABLE_RTT
+				if (rtt_enabled)
+					poll_rtt(cur_target);
+#endif
+			}
+
+			SET_IDLE_STATE(true);
+			size_t size = gdb_getpacket(pbuf, BUF_SIZE);
+			// If port closed and target detached, stay idle
+			if (pbuf[0] != 0x04 || cur_target)
```

Comment:
`pbuf[0]` is a char, so this should compare `'\x04'` for type consistency please